Sources/BraintreeVenmo/BTVenmoAppSwitchRedirectURL.swift

```diff
@@ -60,36 +46,48 @@ struct BTVenmoAppSwitchRedirectURL {
         let serializedBraintreeData = try? JSONSerialization.data(withJSONObject: braintreeData)
         let base64EncodedBraintreeData = serializedBraintreeData?.base64EncodedString()
 
-        var appSwitchParameters: [String: Any] = [
-            "x-success": successReturnURL,
-            "x-error": errorReturnURL,
-            "x-cancel": cancelReturnURL,
+        queryParameters = [
+            "x-success": constructRedirectURL(with: returnURLScheme, result: "success"),
+            "x-error": constructRedirectURL(with: returnURLScheme, result: "error"),
+            "x-cancel": constructRedirectURL(with: returnURLScheme, result: "cancel"),
             "x-source": bundleDisplayName,
             "braintree_merchant_id": merchantID,
             "braintree_access_token": accessToken,
             "braintree_environment": environment,
-            "braintree_sdk_data": base64EncodedBraintreeData ?? ""
+            "resource_id": paymentContextID,
+            "braintree_sdk_data": base64EncodedBraintreeData ?? "",
+            "customerClient": "MOBILE_APP"
         ]
+    }
 
-        if let paymentContextID {
-            appSwitchParameters["resource_id"] = paymentContextID
-        }
+    // MARK: - Internal Methods
+
+    func universalLinksURL() -> URL? {
+        let universalLinkURL = URL(string: "https://venmo.com/go/checkout")!
+
+        var urlComponent = URLComponents(url: universalLinkURL, resolvingAgainstBaseURL: false)!
+        urlComponent.percentEncodedQuery = BTURLUtils.queryString(from: queryParameters as NSDictionary)
+
+        return urlComponent.url
+    }
 
-        var components = appSwitchBaseURLComponents()
-        components.percentEncodedQuery = BTURLUtils.queryString(from: appSwitchParameters as NSDictionary)
+    func appSwitchURL() -> URL? {
```

Comment:
Nit - can we call this `urlSchemeURL()` or something similar? "app switch" encompasses both linking types. Was a little confused when reading the tests real quick

Reply:
For sure! Updated: https://github.com/braintree/braintree_ios/pull/1193/commits/9ca06b4307c684f323edcdad3421d21119f58796